packages/volto/src/helpers/FormValidation/validators.ts

```diff
@@ -86,7 +86,7 @@ export const emailValidator = ({ value, formatMessage }: Validator): string => {
 export const isNumberValidator = ({ value, formatMessage }: Validator) => {
   const floatRegex = /^[+-]?\d+(\.\d+)?$/;
   const isValid =
-    typeof value === 'number' && !isNaN(value) && floatRegex.test(value);
+    typeof value === 'string' && !isNaN(+value) && floatRegex.test(value);
```

Comment:
@Tishasoumya-02 I would allow both, string and number. Could you please take a look?